analyses/molecular-subtyping-neurocytoma/01-neurocytoma-subtyping.Rmd

```diff
@@ -0,0 +1,77 @@
+---
+title: "Subtype Neurocytoma tumors as central or extra-ventricular"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Krutika Gaonkar for D3b
+date: 2020
+---
+
+Subtyping conditions as per issue [#805](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/805) are as follows :
+
+ - If primary_site == "Ventricles", then molecular_subtype == CNC (central neurocytoma) 
+ - If primary_site != "Ventricles", then molecular_subtype == EVN (extraventricular neurocytoma)
+
+## Set up 
+
+```{r}
+library(tidyverse)
+```
+
+## Input directories and files
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+histologies_df <- read_tsv(file.path(data_dir, "pbta-histologies.tsv"))
+
+```
+
+## Output directory and file
+
+```{r}
+results_dir <- "results"
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+outputfile <- file.path(results_dir,"neurocytoma_subtyping.tsv")
+```
+
+Checking if `pathology_diagnosis` has "Neurocytoma" in their value, and lets view the current short_histology values
+
+```{r}
+neurocytoma_from_pathology <- histologies_df %>%
+  filter(str_detect(pathology_diagnosis, "Neurocytoma")
+                    ) 
+  
+neurocytoma_from_pathology$short_histology
+```
+
+There's a "CNS neuroblastoma" in short_histology let's check if pathology_diagnosis and pathology_free_text_diagnosis agree
+
+```{r}
+neurocytoma_from_pathology %>%
+  filter(short_histology == "CNS neuroblastoma"
+                    ) %>%
+  select(pathology_diagnosis,pathology_free_text_diagnosis)
+```
+
+
+Both pathology_diagnosis and pathology_free_text_diagnosis is Neurocytoma so keeping it in Neurocytoma list
```

Comment:
Hi @kgaonkar6 thanks for this! Few comments and suggestions here:
1. I would suggest taking the search for `short_histology` out of this completely, as we are now relying solely on `pathology_diagnosis` and `pathology_free_text_diagnosis` for subtyping. Nice, check though :).
2. How many samples contain `Neurocytoma` in `pathology_diagnosis`? You are also assessing `pathology_free_text_diagnosis` for samples which have `Neurocytoma` in `pathology_diagnosis`. Are there any Neurocytoma samples in `pathology_free_text_diagnosis` that are not listed in `pathology_diagnosis`? (I checked and there is a 1:1 relationship here, but would be a good check to make sure we don't expect any in that `pathology_free_text_diagnosis` that are not in `pathology_diagnosis`.
3. Can you print the table of all primary site just to be sure there are no multiples containing `Ventricles;Other site` for example, in which case the != would not work. I know the answer, just making sure we are justifying steps as we go :).

Reply:
I've added the checks for primary_site values and checking if there are any "Neurocytoma" samples in pathology_free_text_diagnosis that are not listed in pathology_diagnosis.
Thanks!

Reply:
looks good!